### PR TITLE
Integrate `f1_cash_flow` FERC1 table

### PR DESCRIPTION
**Pull request by cmgosnell**
# PR Overview
This PR integrates the `cash_flow_ferc1` table. It was mostly pretty straightforward. A few quirks/questions:
- **What to name the $s?**: Related to #2179. I named it 🥁 ... `amount`, which I don't love.
- **What to do with the start/end balances?**: I took inspiration from @zaneselvans  to just remove the starting balance. But it might not be appropriate for this table. There is ONE `amount_type` that has a starting/ending balance. And the DBF table has net change of this same `amount_type`. I believe one of the breakout XBRL tables contain this net change.
- **Weird row_number == 0**: There are two basically all null dbf records where the row_number is 0. We could deal with this in two way: a) add a row into `dbf_to_xbrl.csv` OR b) add a `targeted_drop_duplicates_dbf`. I went with a) because it felt more straightforward.

# PR Checklist

- [x] Merge the most recent version of the branch you are merging into (probably `dev`).
- [x] Make sure you've included good docstrings.
- [x] Defensive data quality/sanity checks in analyses & data processing functions.
- [x] Update the [release notes](https://catalystcoop-pudl.readthedocs.io/en/latest/release_notes.html) and reference reference the PR and related issues.
- [x] Do your own explanatory review of the PR to help the reviewer understand what's going on and identify issues preemptively.
- [x] All CI checks are passing. [Run tests locally to debug failures](https://catalystcoop-pudl.readthedocs.io/en/latest/dev/testing.html#running-tests-with-tox)
- [ ] Include unit tests for new functions and classes.
- [ ] For major data coverage & analysis changes, [run data validation tests](https://catalystcoop-pudl.readthedocs.io/en/latest/dev/testing.html#data-validation)

### CODE REVIEW

File: src/pudl/package_data/ferc1/dbf_to_xbrl.csv [hunk 1]
```diff
@@ -3757,12 +3758,12 @@ f1_unrcvrd_cost,47,1994,Line 47,,,,
...
-f1_utltyplnt_smmry,4,1994,Other (Specify) 1,,ferc:OtherUtilityMember,,
-f1_utltyplnt_smmry,5,1994,Other (Specify) 2,,ferc:OtherUtility3Member,,
-f1_utltyplnt_smmry,6,1994,Other (Specify) 3,,ferc:OtherUtility2Member,,
-f1_utltyplnt_smmry,7,1994,Common,,ferc:CommonUtilityMember,,
+f1_utltyplnt_smmry,2,1994,Electric,,ferc:_electric_utility_member,,
+f1_utltyplnt_smmry,3,1994,Gas,,ferc:_gas_utility_member,,
+f1_utltyplnt_smmry,4,1994,Other (Specify) 1,,ferc:_other_utility_member,,
+f1_utltyplnt_smmry,5,1994,Other (Specify) 2,,ferc:_other_utility3_member,,
+f1_utltyplnt_smmry,6,1994,Other (Specify) 3,,ferc:_other_utility2_member,,
+f1_utltyplnt_smmry,7,1994,Common,,ferc:_common_utility_member,,
```
Comment (cmgosnell) [inline src/pudl/package_data/ferc1/dbf_to_xbrl.csv:3766]: this is why the builds broke! will revert. didn't mean to do that.

File: src/pudl/package_data/ferc1/dbf_to_xbrl.csv [hunk 2]
```diff
@@ -515,15 +515,15 @@ f1_accumdepr_prvsn,4,1994,(413) Exp. of Elec. Plt. Leas. to Others,reported_valu
 f1_accumdepr_prvsn,5,1994,Transportation Expenses-Clearing,reported_value,transportation_expenses_clearing,,
 f1_accumdepr_prvsn,6,1994,Other Clearing Accounts,reported_value,other_clearing_accounts,,
 f1_accumdepr_prvsn,7,1994,Other Accounts (Specify):,reported_value,other_accounts,,
-f1_accumdepr_prvsn,8,1994,Line 8,reported_value,other_accounts_details,TRUE,DBF contains unstructured data with a description in the item column.
+f1_accumdepr_prvsn,8,1994,Line 8,reported_value,other_accounts_details,True,DBF contains unstructured data with a description in the item column.
```
Comment (cmgosnell) [inline src/pudl/package_data/ferc1/dbf_to_xbrl.csv:518]: there are several scattered changes throughout this CSV: `TRUE` -> `True` and `#N/A` -> `""`

this happened bc I used the XBRL taxonomy viewer thing to do the mapping bc it is way easier/less error prone imo. But it is all in TitleCase, so I read in the csv, replaced with snake_case and saved.

**Review by zaneselvans — CHANGES_REQUESTED** [review]
Structurally, I think this really is a starting_balance + deltas = ending_balance table, but more complex than we've seen before. I think probably everything should end up in a single column with `starting_balance` and `ending_balance` as `value_type`s, such that if you sum up all of the non-calculated (total/net) cash flow changes, and add to starting balance, you get ending balance.

File: src/pudl/metadata/fields.py [hunk 3]
```diff
@@ -62,6 +62,7 @@
         "type": "number",
         "unit": "USD",
     },
+    "amount_type": {"type": "string", "description": "Type of amount reported."},
```
[the remark this answers is not in the document]
Comment (cmgosnell) [reply src/pudl/metadata/fields.py:65]: 100%!

File: src/pudl/transform/ferc1.py [hunk 4]
```diff
@@ -1060,7 +1064,6 @@ def unstack_balances_to_report_year_instant_xbrl(
             )
         return df
 
-    @cache_df(key="xbrl")
```
[the remark this answers is not in the document]
Comment (cmgosnell) [reply src/pudl/transform/ferc1.py:1063]: lol

File: src/pudl/package_data/ferc1/dbf_to_xbrl.csv (hunk 2 above)
[the remark this answers is not in the document]
Comment (cmgosnell) [reply src/pudl/package_data/ferc1/dbf_to_xbrl.csv:518]: see my comment above (i snuck it in after you started this review about why these changes showed up

File: src/pudl/metadata/fields.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (zaneselvans) [reply src/pudl/metadata/fields.py]: If it's going to be super generic, I think explaining why in the description would be helpful. Make it clear that it's for a wide variety of financial accounting quantities reported in FERC 1?

**Review by zaneselvans — APPROVED** [review]

**Pull request merged**